components/compliance-service/inspec/cli.go

```diff
@@ -78,22 +94,41 @@ func Scan(paths []string, target *TargetConfig, timeout time.Duration, env map[s
 	target.ResultIncludeBacktrace = false
 	target.ResultMessageLimit = ResultMessageLimit
 
-	env["HOME"] = os.Getenv("HOME")
+	//env["HOME"] = os.Getenv("HOME")
 
-	args := append([]string{binName, "exec"}, paths...)
+	//args := append([]string{binName, "exec"}, paths...)
 
 	// write the inputs to a file to be passed to inspec during command execution
+	tmpDirPath := fmt.Sprintf("/tmp/inspec-upload-%v", makeTimestamp())
+	args, err := getFirejailArgsaAndOutputFileForExec(fireJailExecProfilePath, paths, tmpDirPath)
+	if err != nil {
+		return nil, nil, NewInspecError(INVALID_PARAM, err.Error())
+	}
+
+	tmpKeys := copyKeyFilesIntoTmpDirectory(tmpDirPath, target.KeyFiles)
+	newTarget := target
+	newTarget.KeyFiles = tmpKeys
+	stdoutFile, erroutFile, shellFile := shellscriptAndResponse(exec_command, tmpDirPath)
+	commandArgs := args
+	commandArgs = append(commandArgs, []string{"/bin/sh", shellFile}...)
+	commandArgs = append(commandArgs, paths...)
 	if len(inputs) > 0 {
-		filename := "/tmp/inputs.yml"
+		filename := path.Join(tmpDirPath, "inputs.yml")
 		err := writeInputsToYmlFile(inputs, filename)
 		if err != nil {
 			errString := fmt.Sprintf("unable to write inputs to file for scan job %s : %s", target.Hostname, err.Error())
-			return nil, nil, NewInspecError(INVALID_PARAM, errString)
+			return nil, nil, NewInspecError(UNKNOWN_ERROR, errString)
 		}
 		args = append(args, fmt.Sprintf("--input-file %s", filename))
 	}
+	//Changing the home directory to tmp directory created
+	env["HOME"] = tmpDirPath
 
-	stdOut, stdErr, err := run(args, target, timeout, env)
+	logrus.Debugf("Run: inspec %v", args)
+	_, _, err = run(commandArgs, newTarget, timeout, env)
```

Comment:
We are not checking this err.

Reply:
this error we are not supposed to check, as we are checking the sucess and json files.